### PR TITLE
Migrate final properties over to `InitializationOptions`.

### PR DESCRIPTION
**Pull request by ckipp01**
This migrates the final remaining server properties over to being
`InitializationOptions`. It should now be fully possibly to configure
the server without the usage of setting the `client` property. The
following properties were migrated.

- `globSyntax` -> `globSyntax`
- `renameFileThreshold` -> `renameFileThreshold`
- `icons` -> `icons`
- `signature-help.command` -> `compilerOptions.parameterHintsCommand`
- `completion.command` -> `compilerOptions.completionCommand`
- `override-def-format` -> compilerOptions.overrideDefFormat`

The following properties were removed as they were no longer used.

- `isWarningsEnabled`
- `bloopEmbeddedVersion`

This is step 1 in #1895 
I'd also like to move around a few things in the new-editor document, but I'm still playing around with how to make it the most clear. However, this should be all of the code changes unless I'm missing something.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/Configs.scala [hunk 1]
```diff
@@ -44,6 +44,11 @@ object Configs {
       new GlobSyntaxConfig(
         System.getProperty("metals.glob-syntax", uri.value)
       )
+    def fromString(value: String): Option[GlobSyntaxConfig] =
+      value match {
+        case "vscode" => Some(vscode)
+        case "uri" => Some(uri)
+      }
```
Comment (kpbochenek) [inline metals/src/main/scala/scala/meta/internal/metals/Configs.scala:51]: case _ => None

File: mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala [hunk 2]
```diff
@@ -6,17 +6,44 @@ package scala.meta.internal.pc
...
+ *                                      populated.
+ * @param isCompletionItemDocumentationEnabled whether the `CompletionItem.documentation`
+ *                                             field should be populated.
+ * @param isCompletionItemResolve whether the client wants Metals to handle `completionItem/resolve`
+ * @param isHoverDocumentationEnabled whether to include docstrings in a `textDocument/hover`.
+ * @param isSignatureHelpDocumentationEnabled whether the `SignatureHelp.documentation` field
+ *                                            shouldbe populated.
+ * @param overrideDefFormat whether the override should include an a unicode icon or only ascii.
+ * @param parameterHintsCommand command identifier to trigger parameter hints.
+ * @param snippetAutoIndent whether the client defaults to adding the indentaiton of the reference
```
Comment (kpbochenek) [inline mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala:21]: indentaiton -> indentation

File: mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala [hunk 3]
```diff
@@ -6,17 +6,44 @@ package scala.meta.internal.pc
...
+ * @param completionCommand command identifier to trigger completion.
+ * @param isCompletionItemDetailEnabled whether the `CompletionItem.detail` field should be
+ *                                      populated.
+ * @param isCompletionItemDocumentationEnabled whether the `CompletionItem.documentation`
+ *                                             field should be populated.
+ * @param isCompletionItemResolve whether the client wants Metals to handle `completionItem/resolve`
+ * @param isHoverDocumentationEnabled whether to include docstrings in a `textDocument/hover`.
+ * @param isSignatureHelpDocumentationEnabled whether the `SignatureHelp.documentation` field
+ *                                            shouldbe populated.
+ * @param overrideDefFormat whether the override should include an a unicode icon or only ascii.
```
Comment (kpbochenek) [inline mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala:19]: an a -> an only?

File: mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala [hunk 4]
```diff
@@ -6,17 +6,44 @@ package scala.meta.internal.pc
...
+ *
+ * @param completionCommand command identifier to trigger completion.
+ * @param isCompletionItemDetailEnabled whether the `CompletionItem.detail` field should be
+ *                                      populated.
+ * @param isCompletionItemDocumentationEnabled whether the `CompletionItem.documentation`
+ *                                             field should be populated.
+ * @param isCompletionItemResolve whether the client wants Metals to handle `completionItem/resolve`
+ * @param isHoverDocumentationEnabled whether to include docstrings in a `textDocument/hover`.
+ * @param isSignatureHelpDocumentationEnabled whether the `SignatureHelp.documentation` field
+ *                                            shouldbe populated.
```
[the remark this answers is not in the document]
Comment (ckipp01) [reply mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala:18]: Fixed.

File: metals/src/main/scala/scala/meta/internal/metals/Configs.scala (hunk 1 above)
Comment (ckipp01) [reply metals/src/main/scala/scala/meta/internal/metals/Configs.scala:51]: Whoops, good catch. Fixed.

File: mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala (hunk 2 above)
Comment (ckipp01) [reply mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala:21]: Fixed.

File: mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala (hunk 3 above)
Comment (ckipp01) [reply mtags/src/main/scala/scala/meta/internal/pc/CompilerInitializationOptions.scala:19]: A yea, that's confusing. Just wanted an `a` there. Fixed.

**Comment by ckipp01** [comment]
Ubuntu on jdk8 seems to be just hanging for some reason. It won't even let me look at the logs. However, everything else passed, so I'm going to go ahead and merge. Thanks for the review @kpbochenek!

I'll follow this up with another pr after getting feedback on [this](https://github.com/scalameta/metals/issues/1895#issuecomment-657195445) to clear up the documentation. So if you have a sec, give me your thoughts there.

**Pull request merged**